app/src/components/TextInput.tsx

```diff
@@ -5,70 +5,79 @@ import "./Components.css";
 
 class TextInput extends EnhancedComponent<ITextInputProps, ITextInputState> {
 
-	public static defaultProps: ITextInputProps = {
-		...EnhancedComponent.defaultProps,
-		defaultText: "",
-		width: "100%",
-		color: "transparent",
-		colorMargin: 0,
-		border: false,
-		borderColor: "black",
-		fontSize: 16,
-	};
+    public static defaultProps: ITextInputProps = {
+        ...EnhancedComponent.defaultProps,
+        defaultText: "",
+        width: "100%",
+        color: "transparent",
+        colorMargin: 0,
+        border: false,
+        borderColor: "black",
+        fontSize: 16,
+		submit: event => {},
+    };
 
-	protected constructor(props: ITextInputProps) {
-		super(props);
-		this.state = {
-			...this.state,
-		};
-		this.updateText = this.updateText.bind(this);
-	}
+    protected constructor(props: ITextInputProps) {
+        super(props);
+        this.state = {
+            ...this.state,
+        };
+        this.updateText = this.updateText.bind(this);
+    }
 
-	private updateText(event: any): void {
-		this.setState({
-			...this.state,
-			text: event.target.value,
-		});
-	}
+    protected updateText(event: any): void {
+        this.setState({
+            ...this.state,
+            text: event.target.value,
+        });
+    }
 
-	public render(): ReactNode {
-		return (
-			<div
-				className="text_input_div"
-				style={{
+    public render(): ReactNode {
+		let input_width = 2 * this.props.colorMargin + 5;
+        return (
+            <div
+                className="text_input_div"
+                style={{
+                    backgroundColor: this.props.color,
+                    outline: "none",
+                    border: Number(this.props.border),
 					width: this.props.width,
-					backgroundColor: this.props.color,
-				}}
-			>
-				<input
-					className="text_input"
-					type="text" name="text_input"
-					placeholder={this.props.defaultText}
-					value={this.state.text}
-					onChange={this.updateText}
-					style={{
-						margin: this.props.colorMargin,
-						fontSize: this.props.fontSize,
-						outline: "none", border: Number(this.props.border)
-					}}
-				/>
-			</div>
-		);
-	}
+                    display: "flex",
+                }}
+            >
+                <form onSubmit={this.props.submit}>
+                    <input
+                        className="text_input"
+                        type="text"
+                        name="text_input"
+                        placeholder={this.props.defaultText}
+                        value={this.state.text}
+                        onChange={this.updateText}
```

Comment:
I think you want to also rerender the parent class in the `onChange` as well, since it's pretty annoying that you have to press enter to see the genres get filtered

Reply:
this is a design choice, not a code issue btw

Reply:
also if you make the change here, then there's no need for that `onKeyDown` enter button thing I mentioned above, since it'll always rerender on a new input value